specification/recoveryservicessiterecovery/resource-manager/Microsoft.RecoveryServices/stable/2018-07-10/service.json

```diff
@@ -14719,6 +14784,39 @@
         }
       }
     },
+    "RecoveryPlanProviderSpecificDetails": {
```

Comment:
Why do you have 2 models for input and response? Couldn't they be the same?

Reply:
we would be sending some other provider specific details as part of response, in future, which may not be part of providerSpecificInput.

Reply:
If you do this, the new response-only fields would just be readonly in the providerSpecificInput. You'd effectively have the 2 models you want with only one type definition.